src/analysis/processing/pdal/qgsalgorithmpdalfilternoiseradius.cpp

```diff
@@ -0,0 +1,105 @@
+/***************************************************************************
+                         qgsalgorithmpdalfilternoiseradius.cpp
+                         ---------------------
+    begin                : December 2025
+    copyright            : (C) 2025 by Jan Caha
+    email                : jan.caha at outlook dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmpdalfilternoiseradius.h"
+
+#include "qgspointcloudlayer.h"
+#include "qgsrunprocess.h"
+
+///@cond PRIVATE
+
+QString QgsPdalFilterNoiseRadiusAlgorithm::name() const
+{
+  return QStringLiteral( "filternoiseradius" );
+}
+
+QString QgsPdalFilterNoiseRadiusAlgorithm::displayName() const
+{
+  return QObject::tr( "Filter Noise (by radius algorithm)" );
+}
+
+QString QgsPdalFilterNoiseRadiusAlgorithm::group() const
+{
+  return QObject::tr( "Point cloud data management" );
+}
+
+QString QgsPdalFilterNoiseRadiusAlgorithm::groupId() const
+{
+  return QStringLiteral( "pointclouddatamanagement" );
+}
+
+QStringList QgsPdalFilterNoiseRadiusAlgorithm::tags() const
+{
+  return QObject::tr( "pdal,lidar,filter,noise,radius" ).split( ',' );
+}
+
+QString QgsPdalFilterNoiseRadiusAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "Filter noise in a point cloud using radius algorithm." );
+}
+
+QString QgsPdalFilterNoiseRadiusAlgorithm::shortDescription() const
+{
+  return QObject::tr( "Filter noise in a point cloud using radius algorithm." );
+}
+
+QgsPdalFilterNoiseRadiusAlgorithm *QgsPdalFilterNoiseRadiusAlgorithm::createInstance() const
+{
+  return new QgsPdalFilterNoiseRadiusAlgorithm();
+}
+
+void QgsPdalFilterNoiseRadiusAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterPointCloudLayer( QStringLiteral( "INPUT" ), QObject::tr( "Input layer" ) ) );
+  addParameter( new QgsProcessingParameterBoolean( QStringLiteral( "REMOVE_NOISE_POINTS" ), QObject::tr( "Remove noise points" ), false ) );
+
+  addParameter( new QgsProcessingParameterNumber( QStringLiteral( "MIN_K" ), QObject::tr( "Minimum number of neighbors in radius" ), Qgis::ProcessingNumberParameterType::Double, 2.0 ) );
+  addParameter( new QgsProcessingParameterNumber( QStringLiteral( "RADIUS" ), QObject::tr( "Standard deviation threshold" ), Qgis::ProcessingNumberParameterType::Double, 1.0 ) );
```

Comment:
Do these parameters accept negative values? If not, it would be nice to specify minimum value.